src/internal_rules.jl

```diff
@@ -718,8 +719,8 @@ function EnzymeRules.reverse(
     ::Const{typeof(cholesky)},
     RT::Type,
     dfact,
-    A;
-    kwargs...)
+    A::Annotation{AT};
+    kwargs...) where {AT <: Array}
```

Comment:
Similarly, maybe use
```suggestion
    A::Annotation{<:Matrix};
    kwargs...)
```
or
```suggestion
    A::Annotation{<:AbstractMatrix};
    kwargs...)
```